### PR TITLE
`Exam mode`: Fix missing repository lock notifications

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/).
- [ ] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [ ] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [ ] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
The "The student repositories for this programming exercise were locked successfully." notification is missing for exam exercises, but it is an important piece of information for instructors to verify if the operation before and after the exam went well.

### Description
The issue was that these type of notifications were "exercise updates" before. In exam exercises, these notifications get filtered out of from the notification bar (since the notification bar is not present during an exam and updates directly get highlighted in the exam viewer).

This PR introduces a new notification type for repository permission updates, which therefore won't be filtered out in the notification bar.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Programming Exercise

1. Press the "lock repositories" / "unlock repositories" buttons and check that a notification is present
2. Wait until after the due date and check that an automatic notification is present.

#### Exam Mode Testing
Prerequisites:
- 1 Instructor
- 1 Exam with a Programming Exercise

1. Press the "lock repositories" / "unlock repositories" buttons and check that a notification is present
2. Wait until after the exam and check that an automatic notification is present.

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2
#### Exam Mode Test
- [x] Test 1
- [x] Test 2

### CODE REVIEW

File: src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html [hunk 1]
```diff
@@ -72,6 +72,7 @@ <h2 *ngIf="isImportFromExistingExercise || isImportFromFile" id="jhi-programming
             ></jhi-programming-exercise-grading>
 
             <jhi-exercise-update-notification
+                *ngIf="!isExamMode"
```
Comment (tobias-lippert) [inline src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html:75]: why?
This is especially relevant for the exam mode

**Review by tobias-lippert — COMMENTED** [review]
Code looks good but I don't understand one client side change

File: src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html (hunk 1 above)
Comment (Strohgelaender) [reply src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html:75]: The client filters out the notifications containing this text and I did not find them to be used at any other place (especially during the exam).

**Review by tobias-lippert — APPROVED** [review]
Code LGTM

**Review by valentin-boehm — APPROVED** [review]
lgtm

**Comment by xHadie** [comment]
_Tested on LS11_
**Automatic notifaction after due date**:
- notifcation was added to notification list and the bell icon showed that a new notification is present, but no growl notifaction bubble popped up.
**Exam mode**
- no notification shows up when exam mode is enabled
- changing the submission policy from "Lock Repository" to either different option and back to "Lock Repository" results in an Internal server error.

**Comment by Strohgelaender** [comment]
@xHadie Thanks for testing.
> but no growl notifaction bubble popped up

I assume you mean the notification on the bottom right of the screen by that. This is intentional, as these notifications should only be used for stuff that needs immediate attention, whereas these types of notifications are more a confirmation that don't require any manual action.

>no notification shows up when exam mode is enabled
Do you mean while working on an exam? This is also correct, as we don't want to distract users while using the exam mode.

> changing the submission policy from "Lock Repository" to either different option and back to "Lock Repository" results in an Internal server error.
This is unrelated to this PR, please open a GitHub issue to track this.

**Review by tobias-lippert — APPROVED** [review]
Reapprove after merge

**Review by valentin-boehm — APPROVED** [review]
reapprove

**Review by RY997 — DISMISSED** [review]
Tested on ts2, worked as expected

**Review by sarpsahinalp — COMMENTED** [review]
Tested on ts2. Seems to be working

File: src/main/webapp/i18n/en/notification.json [hunk 2]
```diff
@@ -52,6 +52,8 @@
                 "programmingTestCasesChanged": "Test cases for programming exercise changed",
                 "newManualFeedbackRequest": "A new manual feedback request has been submitted",
                 "duplicateTestCase": "Duplicate test case was found.",
+                "repositoryLocks": "Permissions Changed",
+                "buildRun": "Build Run Update",
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/en/notification.json:56]: The other strings don't use title case here, is that on purpose?

File: src/main/webapp/i18n/en/notification.json [hunk 3]
```diff
@@ -52,6 +52,8 @@
                 "programmingTestCasesChanged": "Test cases for programming exercise changed",
                 "newManualFeedbackRequest": "A new manual feedback request has been submitted",
                 "duplicateTestCase": "Duplicate test case was found.",
+                "repositoryLocks": "Permissions Changed",
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/en/notification.json:55]:
```suggestion
                "repositoryLocks": "Repository Permissions Changed",
```

File: src/main/webapp/i18n/de/notification.json [hunk 4]
```diff
@@ -52,6 +52,8 @@
                 "programmingTestCasesChanged": "Testfälle für Programmieraufgaben geändert",
                 "newManualFeedbackRequest": "Eine neue Anfrage zu manuellem Feedback wurde eingereicht",
                 "duplicateTestCase": "Doppelte Testfälle wurden gefunden.",
+                "repositoryLocks": "Zugriffsrechte aktualisiert",
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/de/notification.json:55]:
```suggestion
                "repositoryLocks": "Repository-Berechtigungen aktualisiert",
```

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

**Review by JanaNF — APPROVED** [review]
Manual tested on legacy_ts11. Worked as expected.

File: src/main/webapp/i18n/en/notification.json (hunk 2 above)
Comment (JohannesStoehr) [reply src/main/webapp/i18n/en/notification.json]: I'd prefer the titles to be consistent with their capitalization.
Maybe we should think about shortening all of them and putting important info into the text instead the title

**Review by JohannesStoehr — APPROVED** [review]
Code

**Review by JohannesStoehr — APPROVED** [review]

**Review by tobias-lippert — APPROVED** [review]
Code LGTM

**Review by valentin-boehm — APPROVED** [review]
Re-tested locally, works as expected

**Review by maximiliansoelch — APPROVED** [review]
Code looks good.
Tested on legacy ts1, works as expected.

**Pull request merged**